frame/contracts/src/exec.rs

```diff
@@ -903,11 +905,21 @@ where
 			// Every non delegate call or instantiate also optionally transfers the balance.
 			self.initial_transfer()?;
 
+			#[cfg(feature = "unsafe-debug")]
+			let (code_hash, input_clone) = {
+				let code_hash = *executable.code_hash();
+				T::Debug::before_call(&code_hash, entry_point, &input_data);
+				(code_hash, input_data.clone())
+			};
+
 			// Call into the Wasm blob.
 			let output = executable
 				.execute(self, &entry_point, input_data)
 				.map_err(|e| ExecError { error: e.error, origin: ErrorOrigin::Callee })?;
 
+			#[cfg(feature = "unsafe-debug")]
+			T::Debug::after_call(&code_hash, entry_point, &input_clone, &output);
```

Comment:
Let's move in the input as we don't need it anymore. In case the observer wants to store it there is no clone necessary.

```suggestion
			T::Debug::after_call(&code_hash, entry_point, input_clone, &output);
```

Reply:
changed; however, this made `before_call` and `after_call` arguments differ, but I think it might make sense to move input after call